eth/downloader/peer.go

```diff
@@ -409,7 +419,12 @@ func (ps *peerSet) BodyIdlePeers() ([]*peer, int) {
 	idle := func(p *peer) bool {
 		return atomic.LoadInt32(&p.blockIdle) == 0
 	}
-	return ps.idlePeers(62, 64, idle)
+	throughput := func(p *peer) float64 {
```

Comment:
There are ways to specify the field to access. For example, you can pass the throughput functions as a function parameter. See https://golang.org/ref/spec#Function_types

In fact, that is what is done here already, as e.g. `throughput` is passed to `idlePeers`.

One can also specify the field to access as a constant, using e.g. https://golang.org/ref/spec#Constant_declarations though this may be considered less "clean".